File: 백준/Silver/7115. Circle/Circle.c
#include <stdio.h>

int dist2(int x, int y) {
    return x * x + y * y;
}

int main() {
    int k, r; scanf("%d %d", &k, &r);
    int x = 0;
    int y = (r - 1) / k;
    double p = (double)(r) / k;
    p *= p;
    int cnt = 4;
    while (y > x) {
        cnt += 8;
        if (p > dist2(x + 1, y))
            x += 1;
        else if (dist2(x, y - 1) < p && p < dist2(x + 1, y))
            y -= 1;
        else {
            x += 1;
            y -= 1;
        }
    }
    printf("%d", cnt);
    return 0;
}

File: 백준/Silver/7115. Circle/Circle.py
dist2 = lambda x, y: x * x + y * y

k, r = map(int, input().split())
x1, y1 = 0, (r - 1) // k
x2 = y2 = int(r / (k * 2**.5))
p = (r/k)**2
cnt = 4
while (x1, y1) != (x2, y2):
    cnt += 8
    if p > dist2(x1 + 1, y1):
        x1 += 1
    elif dist2(x1, y1 - 1) < p < dist2(x1 + 1, y1):
        y1 -= 1
    else:
        x1 += 1
        y1 -= 1
print(cnt)